### DISCUSSION TITLE
Get the nth element after filtering

### DISCUSSION THREAD

**Discussion by pluma9** [johnkerl/miller#1542] [Q&A] [answered]
Given csv:

```
id,color
1,blue
2,green
3,yellow
4,yellow
```

I want to filter the yellow items and get the second one (i.e. the one with id 4).

I tried the following but it does not work:

```
mlr --icsv --opprint filter '$color == "yellow"' then filter 'FNR == 2' items.csv
```

I believe this is because FNR is the row number in the original item list.

How can I achieve this?

**Comment by aborruso** [comment, marked as answer]
A solution by verb could be this one:

```
mlr --csv cat -n -g color then filter '$color=="yellow" && $n==2' input.csv
```

In output you have

```
n,id,color
2,4,yellow
```

You can add `cut -x -f n` to remove `n`.

**Reply by aborruso** [reply]
Hi @pluma9 did you try it? Is it what you wanted?

**Reply by pluma9** [reply]
Hi @aborruso, it works like a charm. Thanks very much for your support!
